### DISCUSSION TITLE
Reserve Setting Feature Useful

### DISCUSSION THREAD

**Discussion by BJReplay** [jasonacox/Powerwall-Dashboard#194] [General]
I had a glitch in my control algorithm yesterday which I'm yet to investigate, but long story short, it lost access to the solar forecast for today until around 7am so tried to charge my powerwall because it thought we'd have no sun today.

I noticed this last night around 9:45pm and manually set the reserve back to 5%, but at midnight, the control algorithm kicked over a new day, and started to do its thing again.

This is a great example of this trace on the default graph being really quite useful, so I thought I'd share.

As an aside, this is from a newly established "cloud" install of the gateway, which I'll write up.  But that write up is in a queue (after writing up MySQL).  But long story short, I have a way of getting Powerwall-Gateway up and running in the cloud for free, if you're running OpenWRT on your router at home, and don't have suitable hardware to run it on.

![image](https://user-images.githubusercontent.com/37993507/221386448-65b91d27-9506-461a-8ca6-7207d5a38468.png)

**Comment by jasonacox** [comment]
Love this!  And yes, I'm loving having the reserve setting on the graph too. 

Today we have 100% clouds with rain and snow (in Los Angeles!!!) so my [super simple script](https://github.com/jasonacox/pypowerwall/blob/main/tools/cron.sh) saw the 100% cloud coverage and raised reserve to 80% during typical solar production, which activated charging so we aren't using grid power during peak.  It drops the reserve to 20% to activate discharging at 4pm when peak TOU hits.

<img width="661" alt="image" src="https://user-images.githubusercontent.com/836718/221388096-499024d3-3317-4ea7-8c24-5838fe32c60f.png">
